snuba/query/parser/expressions.py

```diff
@@ -15,28 +15,33 @@
 from snuba.query.parser.functions import parse_function_to_expr
 from snuba.query.parser.strings import parse_string_to_expr
 from snuba.util import is_function
+from snuba.query.dsl import plus, multiply
+
 
 minimal_clickhouse_grammar = Grammar(
     r"""
 # This root element is needed because of the ambiguity of the aggregation
 # function field which can mean a clickhouse function expression or the simple
 # name of a clickhouse function.
-root_element    = function_call / function_name
-expression      = function_call / simple_term
-parameters_list = parameter* (expression)
-parameter       = expression space* comma space*
-function_call   = function_name open_paren parameters_list? close_paren (open_paren parameters_list? close_paren)?
-simple_term     = quoted_literal / numeric_literal / column_name
-literal         = ~r"[a-zA-Z0-9_\.:-]+"
-quoted_literal  = "'" string_literal "'"
-string_literal  = ~r"[a-zA-Z0-9_\.:-]*"
-numeric_literal = ~r"-?[0-9]+(\.[0-9]+)?(e[\+\-][0-9]+)?"
-column_name     = ~r"[a-zA-Z_][a-zA-Z0-9_\.]*"
-function_name   = ~r"[a-zA-Z_][a-zA-Z0-9_]*"
-open_paren      = "("
-close_paren     = ")"
-space           = " "
-comma           = ","
+root_element          = function_call / function_name
+expression            = function_call / arithmetic_expression / simple_term
+arithmetic_expression = simple_term2 (("+" arithmetic_expression) / empty)
```

Comment:
No, the second part isn't optional. The point of "empty" is so that there can be matching with either a) another arithmetic expressions or b) nothing at all. So the "empty" is necessary, as far as I can tell. 

Yes, the current version intends for arithmetic_expression to refer to itself (recursive behavior). In future versions, there is a good chance that will be replaced by regex repetition via the "+" you mentioned.

Reply:
Finally, the spacing issue you mentioned was handled in the latest update. Both 5*5+1 and 5+1*5 are handled. It does not matter which order the multiplication and/or addition operators appear in the expression. 

Order of operations with parentheses (for e.g. 3+4*(2-1) ) is a future step.